Exiled.Events/Patches/Events/Scp096/TryingNotToCry.cs

```diff
@@ -25,28 +27,43 @@ namespace Exiled.Events.Patches.Events.Scp096
     [HarmonyPatch(typeof(Scp096), nameof(Scp096.TryNotToCry))]
     internal static class TryingNotToCry
     {
-        private static readonly int DoorMask = LayerMask.GetMask("Door");
-
-        private static bool Prefix(Scp096 __instance)
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
         {
-            if (!Physics.Raycast(__instance.Hub.PlayerCameraReference.position, __instance.Hub.PlayerCameraReference.forward, out RaycastHit hitInfo, 1f, DoorMask))
-                return false;
+            var newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // The index offset.
+            const int offset = 0;
 
-            DoorVariant componentInParent = hitInfo.collider.gameObject.GetComponentInParent<DoorVariant>();
-            if (componentInParent == null || componentInParent.GetExactState() > 0.0 || Scp096._takenDoors.ContainsKey(__instance.Hub.gameObject))
-                return false;
+            // Search for last "lfsfld".
+            var index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldsfld) + offset;
 
-            var ev = new TryingNotToCryEventArgs(__instance, API.Features.Player.Get(__instance.Hub), componentInParent);
+            // Get the count to find the previous index
+            var oldCount = newInstructions.Count;
 
-            Handlers.Scp096.OnTryingNotToCry(ev);
+            // Get the return label from the instruction.
+            var returnLabel = newInstructions[index - 1].labels[0];
 
-            if (ev.IsAllowed)
+            newInstructions.InsertRange(index, new[]
```

Comment:
Same here, add some comments.

Reply:
Added C# code as comments that explain what the IL does. Like [ChangingAttachments](https://github.com/Exiled-Team/EXILED/blob/2.8.0/Exiled.Events/Patches/Events/Item/ChangingAttachments.cs#L48-L56)